### DISCUSSION TITLE
AutomationConditions not working as expected

### DISCUSSION THREAD

**Discussion by scottnoidea** [dagster-io/dagster#23740] [Q&A]
Please help me to understand the AutomationCondition logic below.

I have a time window partitioned dbt asset set to AutomationCondition.any_downstream_conditions().  Its update should be triggered by a couple of downstream on_cron() assets, but it's not working as expected.

It has a handful of upstream assets, some which are static partitioned.  One of these is highlighted in yellow in the policy evaluation below:

![image](https://github.com/user-attachments/assets/664c6e2e-3a22-407d-8a51-93633242f788)

The highlighted asset has been newly updated since the cron tick, however the top level for that condition (highlighted) isn't resolving to true.  It seems to me to satisfy the OR condition.  Forcing re-materializations of the static assets (or recording materialization events) doesn't seem to trigger the condition either.  This is preventing the "all parents updated since" condition from resolving and stopping the materialisation of the asset.

This isn't a problem for the upstream assets which aren't static partitioned - the logic makes sense to me for these and appears to be working OK.

Can you help me understand what is going on?

**Comment by OwenKephart** [comment]
Hi @scottincrypto, apologies for the late response, looks like this slipped through the cracks.

Are you still experiencing this issue on the latest version of dagster, and if so are you able to produce a reproduction of this? The behavior you show in the screenshot is definitely confusing, but a fair number of changes have been made since then which may have ended up resolving this. At the very least, I can confirm that a simple example with upstream static partitioned assets appears to function as expected (waiting for all upstream static partitions to be materialized, then launching the downstream).

**Reply by scottnoidea** [reply]
Thanks - this was in 1.8.0.  I have hacked around the problem since but will try to replicate with a newer version.  Appreciate the followup, thanks.

**Reply by scottnoidea** [reply]
OK, I've had a chance to test this again with 1.8.8 - same behaviour as before:

![image](https://github.com/user-attachments/assets/a6555601-5fca-48b6-bf45-1bed310dc270)

The highlighted check seems to pass the newly updated check, but the asset returns false for the check which blocks the downstream dbt materialisation.

**Reply by OwenKephart** [reply]
Hi @scottincrypto -- one thing to dig into here is that the highlighted row represents the intersection of the  _child partitions_ of the `5 True` parent partitions with the "candidate partitions" (which in this case would just be the latest time partition of the asset).

So this could mean that some parent partitions were materialized, but none of these were parents of the latest time window partition. However, if that highlighted asset is static-partitioned, then the default partition mapping would be an "AllPartitionMapping", which would result in all child partitions being downstream of all upstream partitions, so we wouldn't observe this behavior.

So some follow-up questions:
1. what are the specific partitions definitions of the upstream / downstream assets?
2. is there any custom partition mapping between these assets?
3. if you hover over the `5 True` on the row below the highlighted one, what partition keys was this true for?

**Comment by CainDelta** [comment]
@scottincrypto how did you get it to the name of the dep and not DepConditionWrappCondition ?

**Reply by OwenKephart** [reply]
Hi @CainDelta -- the fact that `DepConditionWrapperCondition` is showing up in place of the asset name is a bug that will be fixed in today's release!

**Reply by CainDelta** [reply]
a happy coincidence , cheers for the prompt reponse @OwenKephart
